connectors/bigquery/flink-connector-gcp-bigquery/src/main/java/com/google/flink/connector/gcp/bigquery/BigQueryCatalog.java

```diff
@@ -0,0 +1,396 @@
+    }
+
+    @Override
+    public void dropDatabase(String databaseName, boolean ignoreIfNotExists, boolean cascade) throws DatabaseNotExistException, DatabaseNotEmptyException, CatalogException {
+        throw new UnsupportedOperationException("Function dropDatabase not supported yet.");
+    }
+
+    @Override
+    public void alterDatabase(String name, CatalogDatabase newDatabase, boolean ignoreIfNotExists) throws DatabaseNotExistException, CatalogException {
+        throw new UnsupportedOperationException("Function alterDatabase not supported yet.");
+    }
+
+    // -- Table Operations --
+    @Override
+    public List<String> listTables(String databaseName) throws DatabaseNotExistException, CatalogException {
+        List<String> targetReturnList = new ArrayList<>();
+        try {
+            DatasetId datasetId = DatasetId.of(this.projectId, databaseName);
+            Page<Table> tables = bigqueryclient.client.listTables(datasetId, TableListOption.pageSize(100));
+
+            tables.iterateAll().forEach(table -> targetReturnList.add(String.format("%s, (Type: %s)",
+                    table.getTableId().getTable(), table.getDefinition().getType().name())));
+
+            return targetReturnList;
+
+        } catch (BigQueryException e) {
+            return List.of();
+        }
+    }
+
+    @Override
+    public List<String> listViews(String databaseName) throws DatabaseNotExistException, CatalogException {
+        List<String> targetReturnList = new ArrayList<>();
+        try {
+            DatasetId datasetId = DatasetId.of(this.projectId, databaseName);
+            Page<Table> tables = bigqueryclient.client.listTables(datasetId, TableListOption.pageSize(100));
+
+            if (tables != null) {
+                tables.iterateAll().forEach(table -> {
+                    if (table.getDefinition().getType() == TableDefinition.Type.VIEW) {
+                        targetReturnList.add(String.format("Success! View ID: %s ", table.getTableId().getTable()));
+                    }
+                });
+            }
+
+            return targetReturnList;
+        } catch (BigQueryException e) {
+            return List.of();
+        }
+    }
+
+    @Override
+    public CatalogBaseTable getTable(ObjectPath tablePath) throws TableNotExistException, CatalogException {
+        try {
+            DatasetId datasetId = DatasetId.of(this.projectId, tablePath.getDatabaseName());
+            TableId tableId = TableId.of(datasetId.getDataset(), tablePath.getObjectName());
+            Table table = bigqueryclient.client.getTable(tableId);
+
+            if (table == null) {
+                throw new TableNotExistException(getName(), tablePath);
+            }
+
+            org.apache.flink.table.api.Schema translatedSchema
+                    = BigQueryTypeUtils.toFlinkSchema(table.getDefinition().getSchema());
+
+            Map<String, String> options = new HashMap<>();
```

Comment:
Should be a check here to make sure this is actually a BQ table and not a BigLake or Iceberg table (i.e. check for the biglake config object or the externalcatalogtableproperties object). For right now its OK to throw an error if its not (next PR can include the logic to construct, say, an Iceberg connector table)

Reply:
the BQ client wouldnt find the table if it's biglake or iceberg. that's probably easier than checking the objects for now